### PR TITLE
8283340: Remove support for -Dsun.misc.URLClassPath.disableJarChecking

### PR DESCRIPTION
**Pull request by cushon**
This change removes support for `-Dsun.misc.URLClassPath.disableJarChecking`. As discussed in the bug the feature doesn't current work, and only ever supported scenarios where a security manager is installed, so it seems safe to remove.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed
- [ ] Change requires a CSR request to be approved

### Issue
 * [JDK-8283340](https://bugs.openjdk.java.net/browse/JDK-8283340): Remove support for -Dsun.misc.URLClassPath.disableJarChecking

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7861/head:pull/7861` \
`$ git checkout pull/7861`

Update a local copy of the PR: \
`$ git checkout pull/7861` \
`$ git pull https://git.openjdk.java.net/jdk pull/7861/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7861`

View PR using the GUI difftool: \
`$ git pr show -t 7861`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7861.diff">https://git.openjdk.java.net/jdk/pull/7861.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/loader/URLClassPath.java [hunk 1]
```diff
@@ -802,37 +799,20 @@ public Void run() throws IOException {
             }
         }
 
-        /* Throws if the given jar file is does not start with the correct LOC */
-        @SuppressWarnings("removal")
```
Comment (jianglizhou) [inline src/java.base/share/classes/jdk/internal/loader/URLClassPath.java:806]: I noticed the @SuppressWarnings("removal") when looking at the PR. It is added by https://github.com/openjdk/jdk/commit/6765f902505fbdd02f25b599f942437cd805cad1 for [JDK-8266459](https://bugs.openjdk.java.net/browse/JDK-8266459). So looks like the method has been targeted for deprcating/removal.

**Comment by AlanBateman** [comment]
I don't know what the motivation for this change is but there is more to this story and I think will require agreement from the security area before removing it.

File: src/java.base/share/classes/jdk/internal/loader/URLClassPath.java (hunk 1 above)
Comment (cushon) [reply src/java.base/share/classes/jdk/internal/loader/URLClassPath.java:806]: Right, the suppression is for the use of `getSecurityManager` here.

**Comment by cushon** [comment]
> I don't know what the motivation for this change is but there is more to this story and I think will require agreement from the security area before removing it.

Thanks @AlanBateman. The motivation is that I was investigating an unrelated change to this code (a potential performance optimization), and it accidentally removed the logic that's catching and ignoring the `IOException`, and when I investigated this logic it looked unused and seemed like a cleanup opportunity. There don't seem to be any public tests for this feature.

If it's security related I understand there may be other internal tests, and this PR may be a non-starter. I'd be interested in anything you can share about the story here, but if this code needs to stay for security reasons feel free to close the PR.

Do you expect this will still be turned down once the `SecurityManager` remove is complete? Or does it make sense to think about generalize this logic so it works even if there isn't a `SecurityManager` enabled?

**Comment by cushon** [comment]
Also, I belatedly realized that I don't think this is actually a no-op: the exception is never shown to the user, but it will cause the jar to be skipped. It should be possible to write a test to exercise that, so I may look in to contributing that if there's interest.

**Comment by seanjmullan** [comment]
I agree with @AlanBateman. Please don't proceed with this fix until further notice.

**Comment by cushon** [comment]
@seanjmullan are there any updates here? Or is the suggestion to drop the PR, because the code is definitely still needed? Thanks!

**Comment by seanjmullan** [comment]
Yes, sorry for the delay. We need to drop this PR for now. We will re-evaluate if this feature/property is still needed at a later point before the Security Manager is removed.

**Comment by cushon** [comment]
No problem, thanks for the update!

**Pull request closed without merging**